core/src/main/java/com/linecorp/armeria/client/metric/AbstractMetricCollectingClient.java

```diff
@@ -46,13 +47,17 @@
     private final MeterIdPrefixFunction meterIdPrefixFunction;
     @Nullable
     private final BiPredicate<? super RequestContext, ? super RequestLog> successFunction;
+    @Nullable
+    private final DistributionStatisticConfig distributionStatisticConfig;
 
     AbstractMetricCollectingClient(
             Client<I, O> delegate, MeterIdPrefixFunction meterIdPrefixFunction,
-            @Nullable BiPredicate<? super RequestContext, ? super RequestLog> successFunction) {
+            @Nullable BiPredicate<? super RequestContext, ? super RequestLog> successFunction,
+            @Nullable DistributionStatisticConfig distributionStatisticConfig) {
```

Comment:
This isn't nullable. Let's remove `@Nullable`.

Reply:
Let's pass it to the `setup` method in `execute(...)` so we can use it when creating the distribution:
```java
RequestMetricSupport.setup(ctx, REQUEST_METRICS_SET, meterIdPrefixFunction, false,
                           successFunction != null ? successFunction::test
                                                   : ctx.options().successFunction(),
                           distributionStatisticConfig);
```
In `RequestMetricSupport`
```java
private static void onRequest(
        RequestLog log, MeterIdPrefixFunction meterIdPrefixFunction, boolean server,
        SuccessFunction successFunction, DistributionStatisticConfig distStatCfg) { ...}

private static void onResponse(
        RequestLog log, MeterIdPrefixFunction meterIdPrefixFunction, boolean server,
        SuccessFunction successFunction, DistributionStatisticConfig distStatCfg) {
        ...
        new DefaultClientRequestMetrics(distStatCfg);
        ...
        new DefaultServiceRequestMetrics(distStatCfg);
        ...
    }
```